### PR TITLE
Task 1 get api topics

### PR DESCRIPTION
**Pull request by Brums16**
pull request for task 2! get api topics

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const request = require("supertest");
+const data = require("../db/data/test-data");
+const { app, server } = require("../app.js");
+const seed = require("../db/seeds/seed");
+const db = require("../db/connection");
+
+beforeEach(() => seed(data));
+afterAll(() => db.end());
+// afterAll(() => server.close());
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: delete any unused console.logs

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const request = require("supertest");
+const data = require("../db/data/test-data");
+const { app, server } = require("../app.js");
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: we're not using server, take out any unused variables

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+describe("GET /api/topics", () => {
+  test("responds with status code 200", () => {
+    return request(app).get("/api/topics").expect(200);
+  });
+  test("responds with an array of topic objects with a slug property and description property", () => {
+    return request(app)
+      .get("/api/topics")
+      .then((response) => {
+        const { topics } = response.body;
```
Comment (roisinoreilly) [inline __tests__/app.test.js:18]: nicely destructured

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        expect(topics.every((topic) => topic.hasOwnProperty("slug"))).toBe(
```
Comment (roisinoreilly) [inline __tests__/app.test.js:19]: I like .every for this assertion

File: app.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const express = require("express");
+const { getTopics } = require("./controllers/topics-controller");
+
+const app = express();
+app.use(express.json());
```
Comment (roisinoreilly) [inline app.js]: are we using this line yet? take it out until you need to use it. Think about what it's actually doing and which endpoint you'll actually need it

File: app.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+app.route("/api/topics").get(getTopics);
+
+const errorHandler = (err, req, res, next) => {
```
Comment (roisinoreilly) [inline app.js]: I would extract this into a separate file to keep your app neat

File: app.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  res.status(400).send(err.message);
+};
+
+app.use(errorHandler);
+
+// const PORT = process.env.PORT || 3000;
```
Comment (roisinoreilly) [inline app.js]: remove this too

File: controllers/topics-controller.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+const { fetchTopics } = require("../models/topics-model");
+
+exports.getTopics = async (req, res, next) => {
```
Comment (roisinoreilly) [inline controllers/topics-controller.js:3]: nicely done

File: models/topics-model.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+const db = require("../db/connection");
+
+exports.fetchTopics = async () => {
+  const { rows } = await db.query(
+    `
+      SELECT * FROM topics
```
Comment (roisinoreilly) [inline models/topics-model.js]: don't forget semicolons on your queries

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
nice start so far, I've left a few comments for some unnecessary lines of code to remove. You've also included a custom error handler which is great at this point, however it's not tested. Can you please take the lines out I've commented on and include a test for your errors on this endpoint remembering to use red/green/refactor and see your test fail. Once you've done that can you send us another PR :)

File: app.js (hunk 5 above)
Comment (Brums16) [reply app.js]: I guess I need this back when I do a post request?

**Comment by Harrison365** [comment]
Good to merge ✅

**Pull request merged**